src/main/java/net/fabricmc/loader/util/DefaultLanguageAdapter.java

```diff
@@ -47,6 +47,11 @@ private DefaultLanguageAdapter() {
 			throw new LanguageAdapterException(e);
 		}
 
+		if (type == Class.class) {
+			//noinspection unchecked
+			return (T) c;
```

Comment:
If the entrypoint points to a field, it should probably assume the field contains a Class and return that. This returns the containing class unconditionally, which would be unexpected.